pyproject.toml

```diff
@@ -1,6 +1,15 @@
 [tool.black]
 line-length = 100
 
+[tool.pyright]
+pythonPlatform = "Linux"
+pythonVersion = "3.8"
+reportUnnecessaryCast = true
+reportUnnecessaryTypeIgnoreComment = true
+stubPath = "./typings"
+typeCheckingMode = "strict"
+verboseOutput = true
+
```

Comment:
is this required for this PR?

Reply:
Not strictly speaking @cragwolfe. Is there a problem with it? I didn't figure it worth a whole separate PR and long story but it was convenient for me to add it in here.